src/Host/Broker/Impl/ApiErrorResult.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.R.Host.Protocol;
+
+namespace Microsoft.R.Host.Broker {
+    internal sealed class ApiErrorResult : ObjectResult {
+        public ApiErrorResult(HttpResponse response, BrokerApiError error) : base(error) {
+            StatusCode = StatusCodes.Status500InternalServerError;
```

Comment:
Please, no HTTP 500. It's just not the right thing to use for REST API errors. HTTP 400 is the generic catch-all for "server was able to handle your request, but the request is invalid for whatever reason". HTTP 5xx is for "server was unable to handle your request, because it crashed due to a bug or other unforeseen circumstances". 4xx is always a client-induced error, and 5xx is always a server-induced error that client has no control over. So any exception that is caused by invalid input received from the client should be 4xx (usually 400).